tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -198,10 +198,14 @@ o switch
                     long i => sb.Append(i),
                     uint i => sb.Append(i),
                     ulong i => sb.Append(i),
-                    IList<object> objs => FormatList(objs, sb),
-                    IEnumerable<KeyValuePair<string, object>> objDict => FormatDictionary(objDict, sb),
+                    IEnumerable<KeyValuePair<string, JToken>> objDict => FormatDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), sb),
                     IEnumerable<KeyValuePair<string, string>> objDict => FormatDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), sb),
-                    _ => throw new Exception($"Couldn't encode: {o}, type: {o.GetType()}")
+                    IEnumerable<KeyValuePair<string, List<string>>> objDict => FormatDictionary(objDict.Select(x => new KeyValuePair<string, object>(x.Key, x.Value)), sb),
+                    IEnumerable<KeyValuePair<string, object>> objDict => FormatDictionary(objDict, sb),
+                    IList<JToken> objs => FormatList(objs.Select(x => (object)x), sb),
+                    IList<string> objs => FormatList(objs.Select(x => (object)x), sb),
+                    IList<object> objs => FormatList(objs, sb),
+                    _ => sb.Append($"Error: couldn't format type: {o?.GetType()}")
```

Comment:
These changes fix the initial bug, the formatter needs to be able to handle the same cases as the encoder. Also, decided it's not helpful to throw and error here.